compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -957,6 +963,64 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       case ctor: MethodSymbol if ctor.isPrimaryConstructor => ctor
     }
 
+  private def needsFixOrCleanup(template: Template): Boolean = {
+    val body = template.body
+    if (body == null) {
+      false
+    } else {
+      body.exists { member =>
+        val sym = member.symbol
+        if (sym != null) sym.hasAnnotation(WasPublicBeforeTyperClass)
```

Comment:
Are there really `null` `body`s and `symbol`s? Wouldn't they be `EmptyTree` and `NoSymbol`, respectively?